executor/split.go

```diff
@@ -244,48 +287,79 @@ func (e *SplitTableRegionExec) Next(ctx context.Context, _ *chunk.Chunk) error {
 	}
 	regionIDs := make([]uint64, 0, len(splitKeys))
 	for _, key := range splitKeys {
+		failpoint.Inject("mockSplitRegionTimeout", func(val failpoint.Value) {
+			if val.(bool) {
+				time.Sleep(time.Second*1 + time.Millisecond*10)
+			}
+		})
+		if isCtxDone(ctxWithTimeout) {
+			break
+		}
 		regionID, err := s.SplitRegion(key, true)
 		if err != nil {
 			logutil.BgLogger().Warn("split table region failed",
 				zap.String("table", e.tableInfo.Name.L),
 				zap.Error(err))
 			continue
 		}
+		if regionID == 0 {
+			continue
+		}
 		regionIDs = append(regionIDs, regionID)
 
-		failpoint.Inject("mockSplitRegionTimeout", func(val failpoint.Value) {
-			if val.(bool) {
-				time.Sleep(time.Second * 1)
-			}
-		})
-
-		if isCtxDone(ctxWithTimeout) {
-			return errors.Errorf("split region timeout(%v)", e.ctx.GetSessionVars().GetSplitRegionTimeout())
-		}
 	}
+	e.splitRegions = len(regionIDs)
 	if !e.ctx.GetSessionVars().WaitSplitRegionFinish {
 		return nil
 	}
+
+	finishScatterNum := waitScatterRegionFinish(e.ctx, ctxWithTimeout, start, s, regionIDs, e.tableInfo.Name.L, "")
+	e.finishScatterNum = finishScatterNum
```

Comment:
```
	finishScatterNum := waitScatterRegionFinish(e.ctx, ctxWithTimeout, start, s, regionIDs, e.tableInfo.Name.L, "")
	e.finishScatterNum = finishScatterNum
```
Please combine them in 1 line, the code length simply increase by 2...(from 116 to 118)

Reply:
Great, thanks.